text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,1031 @@
+support of OTel Arrow, 3) implement a SQL/Arrow engine in Go (big project). A proof of concept using Datafusion has been
+implemented in Rust and has shown very good results.
+
+We believe that because the Arrow IPC mechanism and data format is intended for zero-copy use, we believe it is possible
+to use Arrow libraries written in other languages, for example within the Golang-based OpenTelemetry Collector.
+
+### Further-integrated compression techniques
+
+ZSTD offers a training mode, which can be used to tune the algorithm for a selected type of data. The result of this
+training is a dictionary that can be used to compress the data. Using this [dictionary](http://facebook.github.io/zstd/#small-data)
+can dramatically improve the compression rate for small batches. This future development will build on both the gRPC
+stream approach used in this proposal and the ability to send a ZSTD dictionary over the OTel Arrow stateful protocol,
+allowing us to train the ZSTD algorithm on the first batches and then update the configuration of the ZSTD
+encoder/decoder with an optimized dictionary.
+
+More advanced lightweight compression algorithms on a per column basis could be integrated to the OTel Arrow
+protocol (e.g. delta delta encoding for numerical columns)
+
+### Choosing row-oriented transport when it is more efficient
+
+The columnar representation is more efficient for transporting large homogeneous batches. The support of a mixed approach
+combining automatically column-oriented and row-oriented batches would allow to cover all scenarios. The development of
+a strategy to automatically select the best data representation mode is an open question.
+
+### Unary gRPC OTel Arrow and HTTP OTel Arrow
+
+The design currently calls for the use of gRPC streams to benefit from OTel Arrow transport.  We believe that some of
+this benefit can be had even for unary gRPC and HTTP requests with large request batches to amortize sending of
+dictionary and schema information.  This remains an area for study.
+
+## Appendix A - Protocol Buffer Definitions
+
+Protobuf specification for an Arrow-based OpenTelemetry event.
+
+```protobuf
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This protocol specifies the services and messages utilized by the OTel Arrow
+// Protocol. It extends the OTLP protocol by representing OTLP entities in a
```

Comment:
Please avoid using "extend OTLP" references. Instead say new "Otel Arrow" protocol.

Reply:
I applied this change to the entire OTEP. I also changed the abstract to make a clear distinction between OTLP and OTelArrow.